### PR TITLE
feat: add http server

### PR DESCRIPTION
**Pull request by ZuLiangWang**
# Which issue does this PR close?
None.

# Rationale for this change
CeresMeta currently only includes grpc server, not http server. But http interface need to be provided for some operation and maintenance operations.

# What changes are included in this PR?
* Import third-party library `httprouter`, it is used for simplify server development. https://github.com/julienschmidt/httprouter
* Implement http server based on `httprouter` and `http` in standard library.

# Are there any user-facing changes?
None.

# How does this change test
Local manual test.

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -92,6 +94,8 @@ type Config struct {
 	PeerUrls            string `toml:"peer-urls" json:"peer-urls"`
 	AdvertiseClientUrls string `toml:"advertise-client-urls" json:"advertise-client-urls"`
 	AdvertisePeerUrls   string `toml:"advertise-peer-urls" json:"advertise-peer-urls"`
+
+	DefaultHTTPPort int `toml:"default-http-port" json:"default-http-port"`
```
Comment (chunshao90) [inline server/config/config.go:98]:
```suggestion
	HTTPPort int `toml:"http-port" json:"http-port"`
```

File: server/config/config.go [hunk 2]
```diff
@@ -243,5 +247,6 @@ func MakeConfigParser() (*Parser, error) {
 	fs.IntVar(&cfg.DefaultClusterReplicationFactor, "default-cluster-replication-factor", defaultClusterReplicationFactor, "replication factor of the default cluster")
 	fs.IntVar(&cfg.DefaultClusterShardTotal, "default-cluster-shard-total", defaultClusterShardTotal, "shard total of the default cluster")
 
+	fs.IntVar(&cfg.DefaultHTTPPort, "default-http-port", defaultHTTPPort, "port of http server")
```
Comment (chunshao90) [inline server/config/config.go:250]:
```suggestion
	fs.IntVar(&cfg.HTTPPort, "http-port", defaultHTTPPort, "port of http server")
```

File: server/server.go [hunk 3]
```diff
@@ -20,6 +21,7 @@ import (
 	"github.com/CeresDB/ceresmeta/server/id"
 	"github.com/CeresDB/ceresmeta/server/member"
 	metagrpc "github.com/CeresDB/ceresmeta/server/service/grpc"
+	http2 "github.com/CeresDB/ceresmeta/server/service/http"
```
Comment (chunshao90) [inline server/server.go:24]: Why use `https`?

File: server/service/http/api.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+func NewAPI(procedureManager procedure.Manager, procedureFactory *procedure.Factory, clusterManager cluster.Manager) *API {
+	return &API{
+		procedureManager: procedureManager,
+		procedureFactory: procedureFactory,
+		clusterManager:   clusterManager,
+	}
+}
+
+func (a *API) NewAPIRouter() *Router {
+	router := New().WithPrefix("/api/v1").WithInstrumentation(printRequestInsmt)
```
Comment (chunshao90) [inline server/service/http/api.go:38]: Define a const.

File: server/service/http/api.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+	router.Post("/transferLeader", a.transferLeader)
+
+	return router
+}
+
+// printRequestInsmt used for printing every request information.
```
Comment (chunshao90) [inline server/service/http/api.go:45]:
```suggestion
// printRequestInsmt is used for printing every request information.
```

File: server/service/http/api.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	ShardID           uint64 `json:"shardID"`
+	NewLeaderNodeName string `json:"newLeaderNodeName"`
+}
+
+// TODO: impl this function
+func (a *API) transferLeader(_ http.ResponseWriter, req *http.Request) {
+	var transferLeaderRequest TransferLeaderRequest
+	err := json.NewDecoder(req.Body).Decode(&transferLeaderRequest)
+	if err != nil {
+		log.Error("decode request body failed")
```
Comment (chunshao90) [inline server/service/http/api.go:109]:
```suggestion
		log.Error("decode request body failed",zap.Error(err))
```

File: server/server.go (hunk 3 above)
Comment (ZuLiangWang) [reply server/server.go:24]: It's not https, golang has already contains http package, and I define another one, import my http package will be auto renamed as http2.

**Review by chunshao90 — APPROVED** [review]
LGTM

**Pull request merged**